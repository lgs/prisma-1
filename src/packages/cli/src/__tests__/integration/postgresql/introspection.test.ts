import * as IntegrationTest from '../../__helpers__/integrationTest'
import { database } from './__database'
import { scenarios } from './__scenarios'

IntegrationTest.introspectionIntegrationTest({ scenarios, database })
