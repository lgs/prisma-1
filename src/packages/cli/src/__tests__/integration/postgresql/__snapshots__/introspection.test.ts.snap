// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`introspection: create with data: datamodel 1`] = `
generator client {
  provider = "prisma-client-js"
  output = "***"
}

datasource postgresql {
  provider = "postgresql"
  url = "***"
}

model teams {
  id   Int    @default(autoincrement()) @id
  name String @unique
}

`;

exports[`introspection: create with data: warnings 1`] = `Array []`;

exports[`introspection: create with empty data and SQL default: datamodel 1`] = `
generator client {
  provider = "prisma-client-js"
  output = "***"
}

datasource postgresql {
  provider = "postgresql"
  url = "***"
}

model teams {
  id   Int    @default(autoincrement()) @id
  name String @default("alice")
}

`;

exports[`introspection: create with empty data and SQL default: warnings 1`] = `Array []`;

exports[`introspection: delete where composite unique: datamodel 1`] = `
generator client {
  provider = "prisma-client-js"
  output = "***"
}

datasource postgresql {
  provider = "postgresql"
  url = "***"
}

model users {
  id    Int    @default(autoincrement()) @id
  email String
  name  String

  @@unique([email, name], name: "users_email_name_key")
}

`;

exports[`introspection: delete where composite unique: warnings 1`] = `Array []`;

exports[`introspection: findMany - email text: datamodel 1`] = `
generator client {
  provider = "prisma-client-js"
  output = "***"
}

datasource postgresql {
  provider = "postgresql"
  url = "***"
}

model users {
  id    Int     @default(autoincrement()) @id
  email String?
}

`;

exports[`introspection: findMany - email text: warnings 1`] = `Array []`;

exports[`introspection: findMany - email varchar(50) not null unique: datamodel 1`] = `
generator client {
  provider = "prisma-client-js"
  output = "***"
}

datasource postgresql {
  provider = "postgresql"
  url = "***"
}

model users {
  id    Int    @default(autoincrement()) @id
  email String @unique
}

`;

exports[`introspection: findMany - email varchar(50) not null unique: warnings 1`] = `Array []`;

exports[`introspection: findMany where contains and boolean: datamodel 1`] = `
generator client {
  provider = "prisma-client-js"
  output = "***"
}

datasource postgresql {
  provider = "postgresql"
  url = "***"
}

model posts {
  id        Int     @default(autoincrement()) @id
  title     String
  published Boolean @default(false)
}

`;

exports[`introspection: findMany where contains and boolean: warnings 1`] = `Array []`;

exports[`introspection: findMany where unique: datamodel 1`] = `
generator client {
  provider = "prisma-client-js"
  output = "***"
}

datasource postgresql {
  provider = "postgresql"
  url = "***"
}

model users {
  id    Int    @default(autoincrement()) @id
  email String @unique
}

`;

exports[`introspection: findMany where unique: warnings 1`] = `Array []`;

exports[`introspection: findOne where PK with include: datamodel 1`] = `
generator client {
  provider = "prisma-client-js"
  output = "***"
}

datasource postgresql {
  provider = "postgresql"
  url = "***"
}

model posts {
  id      Int    @default(autoincrement()) @id
  user_id Int
  title   String
  users   users  @relation(fields: [user_id], references: [id])
}

model users {
  id    Int     @default(autoincrement()) @id
  email String  @unique
  posts posts[]
}

`;

exports[`introspection: findOne where PK with include: warnings 1`] = `Array []`;

exports[`introspection: findOne where PK with select: datamodel 1`] = `
generator client {
  provider = "prisma-client-js"
  output = "***"
}

datasource postgresql {
  provider = "postgresql"
  url = "***"
}

model teams {
  id    Int    @id
  name  String @unique
  email String @unique
}

`;

exports[`introspection: findOne where PK with select: warnings 1`] = `Array []`;

exports[`introspection: findOne where PK: datamodel 1`] = `
generator client {
  provider = "prisma-client-js"
  output = "***"
}

datasource postgresql {
  provider = "postgresql"
  url = "***"
}

model teams {
  id   Int    @id
  name String @unique
}

`;

exports[`introspection: findOne where PK: warnings 1`] = `Array []`;

exports[`introspection: findOne where composite unique: datamodel 1`] = `
generator client {
  provider = "prisma-client-js"
  output = "***"
}

datasource postgresql {
  provider = "postgresql"
  url = "***"
}

model users {
  id    Int    @default(autoincrement()) @id
  email String
  name  String

  @@unique([email, name], name: "users_email_name_key")
}

`;

exports[`introspection: findOne where composite unique: warnings 1`] = `Array []`;

exports[`introspection: findOne where unique with foreign key and unpack: datamodel 1`] = `
generator client {
  provider = "prisma-client-js"
  output = "***"
}

datasource postgresql {
  provider = "postgresql"
  url = "***"
}

model posts {
  id      Int    @default(autoincrement()) @id
  user_id Int
  title   String
  users   users  @relation(fields: [user_id], references: [id])
}

model users {
  id    Int     @default(autoincrement()) @id
  email String  @unique
  posts posts[]
}

`;

exports[`introspection: findOne where unique with foreign key and unpack: warnings 1`] = `Array []`;

exports[`introspection: findOne where unique: datamodel 1`] = `
generator client {
  provider = "prisma-client-js"
  output = "***"
}

datasource postgresql {
  provider = "postgresql"
  url = "***"
}

model users {
  id    Int    @default(autoincrement()) @id
  email String @unique
}

`;

exports[`introspection: findOne where unique: warnings 1`] = `Array []`;

exports[`introspection: update where composite unique: datamodel 1`] = `
generator client {
  provider = "prisma-client-js"
  output = "***"
}

datasource postgresql {
  provider = "postgresql"
  url = "***"
}

model users {
  id    Int    @default(autoincrement()) @id
  email String
  name  String

  @@unique([email, name], name: "users_email_name_key")
}

`;

exports[`introspection: update where composite unique: warnings 1`] = `Array []`;

exports[`introspection: update where with boolean data and select: datamodel 1`] = `
generator client {
  provider = "prisma-client-js"
  output = "***"
}

datasource postgresql {
  provider = "postgresql"
  url = "***"
}

model teams {
  id     Int     @default(autoincrement()) @id
  name   String  @unique
  active Boolean @default(true)
}

`;

exports[`introspection: update where with boolean data and select: warnings 1`] = `Array []`;

exports[`introspection: update where with boolean data: datamodel 1`] = `
generator client {
  provider = "prisma-client-js"
  output = "***"
}

datasource postgresql {
  provider = "postgresql"
  url = "***"
}

model teams {
  id     Int     @default(autoincrement()) @id
  name   String  @unique
  active Boolean @default(true)
}

`;

exports[`introspection: update where with boolean data: warnings 1`] = `Array []`;

exports[`introspection: update where with numeric data: datamodel 1`] = `
generator client {
  provider = "prisma-client-js"
  output = "***"
}

datasource postgresql {
  provider = "postgresql"
  url = "***"
}

model teams {
  id   Int    @default(autoincrement()) @id
  name String @unique
}

`;

exports[`introspection: update where with numeric data: warnings 1`] = `Array []`;

exports[`introspection: update where with string data: datamodel 1`] = `
generator client {
  provider = "prisma-client-js"
  output = "***"
}

datasource postgresql {
  provider = "postgresql"
  url = "***"
}

model teams {
  id   Int    @default(autoincrement()) @id
  name String @unique
}

`;

exports[`introspection: update where with string data: warnings 1`] = `Array []`;

exports[`introspection: updateMany where with string data - check findMany: datamodel 1`] = `
generator client {
  provider = "prisma-client-js"
  output = "***"
}

datasource postgresql {
  provider = "postgresql"
  url = "***"
}

model teams {
  id   Int    @default(autoincrement()) @id
  name String
}

`;

exports[`introspection: updateMany where with string data - check findMany: warnings 1`] = `Array []`;

exports[`introspection: updateMany where with string data - check returned count: datamodel 1`] = `
generator client {
  provider = "prisma-client-js"
  output = "***"
}

datasource postgresql {
  provider = "postgresql"
  url = "***"
}

model teams {
  id   Int    @default(autoincrement()) @id
  name String
}

`;

exports[`introspection: updateMany where with string data - check returned count: warnings 1`] = `Array []`;
